packages/ui/atoms/Action/Action.ts

```diff
@@ -0,0 +1,80 @@
+import { Base } from '@studiometa/js-toolkit';
+import { isFunction } from '@studiometa/js-toolkit/utils';
+import type { BaseProps, BaseConfig } from '@studiometa/js-toolkit';
+
+export interface ActionProps extends BaseProps {
+  $options: {
+    on: string;
+    target: string;
+    method: string;
+    selector: string;
+  };
+}
+
+/**
+ * Action class.
+ */
+export class Action<T extends BaseProps = BaseProps> extends Base<ActionProps & T> {
+  static config: BaseConfig = {
+    name: 'Action',
+    options: {
+      on: {
+        type: String,
+        default: 'click',
+      },
+      target: String,
+      method: String,
+      selector: String,
+    },
+  };
+
+  /**
+   * Run method on targeted components
+   */
+  doAction() {
+    const { target: componentNames, method, selector } = this.$options;
+
+    let targets = componentNames.includes(' ')
+      ? componentNames
+          .split(' ')
+          .flatMap((componentName) => this.$root.$children?.[componentName] as Base[])
+      : (this.$root.$children?.[componentNames] as Base[]);
+
+    if (!targets.length) {
+      return;
+    }
+
+    if (selector || selector.length > 0) {
+      targets = targets.filter((target) => target.$el.matches(selector));
+    }
+
+    targets.forEach((target) => {
+      if (!isFunction(target[method])) {
+        return;
+      }
+
+      target[method]();
+    });
+  }
+
+  /**
+   * Mounted
+   */
+  mounted() {
+    const { on: eventName, target, method } = this.$options;
+
+    if (!target || !method || target.length <= 0 || method.length <= 0) {
+      this.$terminate();
+      return;
+    }
+
+    this.$el.addEventListener(eventName, this.doAction.bind(this));
+  }
+
+  /**
+   * Destroyed
+   */
+  destroyed() {
+    this.$el.removeEventListener(this.$options.on, this.doAction);
```

Comment:
**issue :** `this.doAction` is not the same value as `this.doAction.bind(this)`, as the `Function.bind()` method returns a new function

I would suggest to rename `doAction` to `handleEvent` which would allow to simplify managing the event listener (see https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#specifying_this_using_bind). 

```suggestion
    this.$el.removeEventListener(this.$options.on, this);
```

You would have the following:

```js
handleEvent() {
  const { target: componentNames, method, selector } = this.$options;
  // ...
}

mounted() {
  // ...
  this.$el.addEventListener(eventName, this);
}

destroyed() {
  this.$el.removeEventListener(this.$options.on, this);
}
```